bazel_codegen/lib/src/arg_parser.dart

```diff
@@ -0,0 +1,130 @@
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:logging/logging.dart';
+
+const _rootDirParam = 'root-dir';
+const _helpParam = 'help';
+const _inExtension = 'in-extension';
+const _logLevelParam = 'log-level';
+const _logPathParam = 'log';
+const _outExtension = 'out-extension';
+const _outParam = 'out';
+const _packagePathParam = 'package-path';
+const _packageMapParam = 'package-map';
+const _srcsParam = 'srcs-file';
+const _summariesParam = 'use-summaries';
+
+// All arguments other than `--help` and `--use-summaries` are required.
+final _argParser = new ArgParser()
+  ..addOption(_rootDirParam,
+      allowMultiple: true,
+      help: 'One or more workspace directories to check when reading files.')
+  ..addFlag(_helpParam,
+      abbr: 'h', negatable: false, help: 'Prints this message and exits')
+  ..addOption(_inExtension, help: 'The file extension to process')
+  ..addOption(_logLevelParam,
+      allowed: _optionToLogLevel.keys.toList(),
+      defaultsTo: 'warning',
```

Comment:
You could do something like this as well to be a bit more robust, although its not really necessary:

```dart
defaultsTo: () {
  for (var k in _optionToLogLevel.keys) {
    if (_optionToLogLevel[k] == Level.WARNING) return k;
  }
  throw 'Unable to find Level.WARNING in `_optionToLogLevel` values';
}(),
```